src/Utilities/functions.jl

```diff
@@ -379,14 +379,11 @@ function Base.getindex(
     terms = MOI.VectorAffineTerm{T}[]
     # assume at least one term per index
     sizehint!(terms, length(I))
-    constant = Vector{T}(undef, length(I))
-    for (i, j) in enumerate(I)
-        for term in it.f.terms
-            if term.output_index == j
-                push!(terms, MOI.VectorAffineTerm(i, term.scalar_term))
-            end
+    constant = it.f.constants[I]
+    for term in it.f.terms
+        if term.output_index in I
+            push!(terms, MOI.VectorAffineTerm(length(terms)+1, term.scalar_term))
```

Comment:
`length(terms)+1` is incorrect, it should be the index in `I`, use `findfirst` instead of `term.output_index in I`

Reply:
yes I thought so, but the bug was already there in the old version:
```julia
append!(terms, map(t -> MOI.VectorAffineTerm(i, t), g.terms))
```
where `i` is the first term of the `enumerate(I)`